flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -610,6 +610,58 @@ object GenerateUtils {
       generateInputFieldUnboxing(ctx, inputType, inputCode, inputCode)
   }
 
+  def generateFieldAccessForCountCol(
+      ctx: CodeGeneratorContext,
+      inputType: LogicalType,
+      inputTerm: String,
+      index: Int): GeneratedExpression = inputType.getTypeRoot match {
+    // ordered by type root definition
+    case ROW | STRUCTURED_TYPE =>
```

Comment:
Do we support count `row` or `structured` type now?

Reply:
> Do we support count `row` or `structured` type now?

Done!